examples/plot_digits.py

```diff
@@ -0,0 +1,181 @@
+# multiple clustered inference procedures. This approach derandomizes the inference
+# procedure and produces more stable results. We use the class:`hidimstat.EnCluDL`
+# for this purpose with 10 bootstraps. While this approach is more computationally intensive than
+# single clustered inference, the procedure can be parallelized across clustering
+# repetitions using the ``n_jobs`` parameter.
+
+
+from sklearn.cluster import FeatureAgglomeration
+from sklearn.feature_extraction import image
+from sklearn.linear_model import LassoCV
+
+from hidimstat import DesparsifiedLasso
+from hidimstat.ensemble_clustered_inference import EnCluDL
+
+fwer = 0.1
+n_jobs = 5
+n_clusters = 100
+
+encludl = EnCluDL(
+    clustering=clustering,
+    desparsified_lasso=DesparsifiedLasso(estimator=LassoCV(max_iter=1000)),
+    n_bootstraps=10,
+    n_jobs=n_jobs,
+    random_state=0,
+    cluster_boostrap_size=0.5,
+)
+
+encludl.fit_importance(X_4_7, y_4_7)
+selected_pos_4_7 = encludl.pvalues_ < fwer / 2 / n_clusters
+selected_neg_4_7 = encludl.one_minus_pvalues_ < fwer / 2 / n_clusters
+
+encludl.fit_importance(X_0_1, y_0_1)
+selected_pos_0_1 = encludl.pvalues_ < fwer / 2 / n_clusters
+selected_neg_0_1 = encludl.one_minus_pvalues_ < fwer / 2 / n_clusters
+
+encludl.fit_importance(X_0_9, y_0_9)
+selected_pos_0_9 = encludl.pvalues_ < fwer / 2 / n_clusters
+selected_neg_0_9 = encludl.one_minus_pvalues_ < fwer / 2 / n_clusters
+
+
+# %%
+# Visualizing the results
+# -----------------------
+# Finally, we visualize the significant pixels identified by EnCluDL for each of the
+# three classification tasks.
+
+import matplotlib.pyplot as plt
+from matplotlib.colors import ListedColormap
+
+_, axes = plt.subplots(1, 3, figsize=(5, 2), subplot_kw={"xticks": [], "yticks": []})
+
+for i, (title, selected_pos, selected_neg) in enumerate(
+    [
+        ("4 vs 7", selected_pos_4_7, selected_neg_4_7),
+        ("0 vs 1", selected_pos_0_1, selected_neg_0_1),
+        ("0 vs 9", selected_pos_0_9, selected_neg_0_9),
+    ]
+):
+    mask_encludl = np.zeros(shape)
+    mask_encludl[selected_pos.reshape(shape)] = 1
+    mask_encludl[selected_neg.reshape(shape)] = -1
+
+    cmap = ListedColormap(["tab:red", "white", "tab:blue"])
+    axes[i].imshow(mask_encludl, cmap=cmap, vmin=-1, vmax=1)
+    axes[i].set_title(title, fontweight="bold", y=1.0)
+
+plt.tight_layout()
+
+
+# %%
+# Performing inference on handwritten digits is challenging because the images are not
+# perfectly aligned, and the pixel regions occupied by digits vary from one image to
+# another. However, using EnCluDL, we can still identify clusters of pixels that are
+# statistically significant for the classification tasks. For example, we can detect
+# the bottom left portion of the loop when distinguishing between digits 0 and 9, the
+# bottom left corner of digit 4 when distinguishing between digits 4 and 7, and the
+# central vertical stroke when distinguishing between digits 0 and 1.
```

Comment:
Just wondering: does the result look nicer with more bootstraps ? If yes, it would be worth doing it.

Reply:
Unfortunatly no. I tried with 50 and it's almost the same. WDYT about this example? Should it be merged?